packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -975,9 +979,9 @@ class AdbLogReader extends DeviceLogReader {
         'flutter',
       ] else if (apiVersion != null && apiVersion >= kLollipopVersionCode) ...<String>[
         // Otherwise, filter for logs appearing past the present.
-        // Empty `-T` means the timestamp of the logcat command invocation.
+        // '-T 0` means the timestamp of the logcat command invocation.
         '-T',
-        device.lastLogcatTimestamp ?? '',
+        if (device.lastLogcatTimestamp != null) '\'${device.lastLogcatTimestamp}\'' else '0',
```

Comment:
Was this change necessary for the fix?

Reply:
Per adb specs, the `-T` has to be followed with either plain number or quoted timestamp.